common/core/desktop/doc/BUILDING.md

```diff
@@ -2,79 +2,154 @@
 
-Ensure you have Python3 correctly installed and can run the command `pip`.
+* Install Python 3:
 
-Then install meson:
+  You can get the official Python installer from the official Python site:
+  <https://www.python.org/downloads/windows/>
+
+  Ensure you have Python3 correctly installed and can run the command `pip`.
+
+  Ensure that Python 3 is on the path ahead of Python 2 (if installed).
+
+* Install meson:
+
+  ```bash
+  python -m pip install meson
+  ```
+
+* Install [ninja](https://ninja-build.org/) 1.10+:
+
+  ```bash
+  choco install ninja
+  ```
+
+* Install Rust:
+
+  After downloading and running the executable from:
+  <https://www.rust-lang.org/tools/install>
+
+  Run the command:
+
+  ```bash
+  rustup target add i686-pc-windows-msvc
+  rustup target add wasm32-unknown-unknown
+  ```
+
+* Install wasm-pack:
+  <https://github.com/rustwasm/wasm-pack/releases/download/v0.9.1/wasm-pack-init.exe>
+
+* Install Emscripten:
+  <https://emscripten.org/docs/getting_started/downloads.html#sdk-download-and-install>
+
+* Add emcc to PATH (probably upstream\enscripten)
+
+#### Windows environment
+
+On Windows you may need to set `SDKVER` environment variable to the current
+Windows SDK version, if it cannot be automatically detected.
 
 ```bash
-python3 -m pip install meson
+export SDKVER=10.0.19041.0
 ```
 
+#### kmcomp
+
 Note on paths for kmcomp:
 
 The search path can be edited through System settings / Advanced system settings
 / Environment Variables / User environment variables.
 
-If you have Keyman Developer installed, add `%KeymanDeveloperPath%` to your
-path. Otherwise, add the path where you extracted the kmcomp archive.
-
-#### Installing Rust
+If you have Keyman Developer installed, kmcomp should be on the path already;
+otherwise add `%KeymanDeveloperPath%` to your path.
 
-After downloading and running the executable from:
-<https://www.rust-lang.org/tools/install>
-
-Run the command:
-```bash
-rustup target add i686-pc-windows-msvc
-```
+If you do not have Keyman Developer installed, add the path where you extracted
+the kmcomp archive.
 
 ### Linux
 
 #### Ubuntu and Debian
 
-You can install `meson` and Rust through the package manager:
+* Install Python
 
-```bash
-sudo apt update
-sudo apt install meson cargo
-rustup target add i686-pc-windows-msvc
-```
+  ```bash
+  sudo apt install python3
+  ```
+
+* Install Meson:
+
+  ```bash
+  sudo apt install meson
+  ```
+
+* Install Rust:
+
+  # TODO
+  ```bash
+  sudo apt install libssl-dev
+  sudo snap install rustup --classic
+  # sudo apt install curl
+  sudo apt install cargo
+  rustup target add wasm32-unknown-unknown
+  ```
+
+* Install wasm-pack:
 
-You'll also need the `kmcomp` wrapper - see below.
+  ```bash
+  curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
+  ```
+
+* Install Enscripten (including adding to path with `emsdk_env.sh`):
+  <https://emscripten.org/docs/getting_started/downloads.html#sdk-download-and-install>
```

Comment:
since there is already a [debian package](https://packages.debian.org/search?keywords=emscripten) for an older version, perhaps it'd be worth upstreaming (contributing) the version bump into debian?

Reply:
I'm not familiar with the debian package details so not sure if this is viable, @ermshiperete? Does it help our scenario?

Reply:
ah, good, I overlooked that there's a package for an older version. Yes, that might be an option.